README.md

```diff
@@ -1 +1,801 @@
+let authResult = transport.send(request: authRequest)
+
+if let sessionId: String = getSessionId(authResult) {
+    let userSearchRequest = HTTPRequest(endpoint: "/user", base: baseRequest)
+        .with(cookieName: "SESSION_ID", value: sessionId)
+        .with(parameters: ["first_name": "John", "last_name": "Appleseed"], encoding: .url)
+
+    let searchResult = transport.send(request: userSearchRequest)
+    if let users: [User] = getUsers(searchResult) {
+        showUsers(users)
+    } else {
+        showEmptyScreen()
+    }
+} else {
+    showError()
+}
+```
+
+<a name="logging" />
+
+### Logging
+
+```swift
+let transport = HTTPTransport(
+    requestInterceptors: [
+        LogRequestInterceptor(logLevel: LogRequestInterceptor.LogLevel.url),
+    ],
+    responseInterceptors: [
+        LogResponseInterceptor(logLevel: LogResponseInterceptor.LogLevel.everything),
+    ]
+)
+
+let result = transport.send(...)
+```
+
+<a name="cookies" />
+
+### Send and receive cookies
+
+```swift
+let cookieStorage: CookieStoring & CookieProviding = getCookieStorage()
+
+let transport = HTTPTransport(
+    requestInterceptors: [
+        AddCookieInterceptor(cookieProvider: cookieStorage),
+    ],
+    responseInterceptors: [
+        ReceivedCookieInterceptor(cookieStorage: cookieStorage),
+    ]
+)
+
+let result = transport.send(...)
+```
+
+<a name="post_parameters" />
+
+### POST request with body & URL parameters
+
+```swift
+let urlParameters = HTTPRequestParameters(
+    parameters: ["first_name" : "John"],
+    encoding: .url
+)
+
+let bodyParameters = HTTPRequestParameters(
+    parameters: ["salary" : 100000],
+    encoding: .json
+)
+
+let updateSalaryRequest = HTTPRequest(
+    httpMethod: HTTPRequest.HTTPMethod.post,
+    endpoint: "https://api.company.com/employees",
+    parameters: [urlParameters, bodyParameters]
+)
+
+let result = transport.send(request: updateSalaryRequest)
+```
+
+<a name="sha1_fingerprint" />
+
+### SSL pinning with SHA1 fingerprint
+
+```swift
+let fingerprint =
+    "ED D6 27 B8 8B 51 B0 24 B9 BF 90 4C D4 AB 9A AB E2 4B 93 00"
+        .replacingOccurrences(of: " ", with: "")
+
+let security = Security(
+    certificates: [
+        TrustPolicyManager.Certificate(host: "google.com", fingerprint: .sha1(fingerprint: fingerprint))
+    ]
+)
+
+let transport = HTTPTransport(security: security)
+let result = transport.send(request: HTTPRequest(endpoint: "https://google.com/ncr"))
+```
+
+<a name="evolution" />
+
+## Evolution
+
+You may have noticed that our library tries not to expose Alamofire interfaces. There is a simple idea to get rid of this transitive
+dependency, and to erect an independent logic on top of the **URLSession** framework.
+
+These far-reaching plans require significant efforts we cannot afford right now. Still, it is a major target we aspire to hit eventually.
+
+So, pull requests are welcome, but consider creating a tentative issue before the actual coding.
```

Comment:
## Appreciation

Inspired by ![RedMadRobot HTTPTransport](https://github.com/RedMadRobot/http-transport). We wanted to implement our vision of HTTPTransport and improve the framework in our own way.